src/FrameBuffer.cpp

```diff
@@ -264,12 +265,19 @@ void FrameBuffer::copyRdram()
 			// This is necessary for auxilary buffers: game can restore content of RDRAM when buffer is not needed anymore
 			// Thus content of RDRAM on moment of buffer creation will be the same as when buffer becomes obsolete.
 			// Validity check will see that the RDRAM is the same and thus the buffer is valid, which is false.
-			// It can be enough to write data just little more than treshold level, but more safe to write twice as much in case that some values in buffer match our fingerprint.
 			const u32 twoPercent = dataSize / 200;
 			u32 start = m_startAddress >> 2;
 			u32 * pData = (u32*)RDRAM;
 			for (u32 i = 0; i < twoPercent; ++i)
-				pData[start++] = m_startAddress;
+			{
+				if (i < 4)
+					pData[start++] = fingerprint[i];
+				else
+					pData[start++] = 0;
```

Comment:
Why modify data below fingerprint if you check fingerprint only?
I mean the code after:
//check if our fingerprint is still there